File: examples/directory/ipClient.py
# Fetch your external IP


import quark_twisted_runtime
import ipify


class Consumer(object):
    def __init__(self, runtime):
        self.runtime = runtime

    def consume(self, ip):
        print "IP is", ip
        self.runtime.reactor.stop()


def main():
    runtime = quark_twisted_runtime.get_runtime()
    ipify.MyExternalIP(runtime, Consumer(runtime))
    runtime.reactor.run()  # Use the default twisted reactor


if __name__ == '__main__':
    main()

File: examples/directory/ipClient.js
// Fake service (makes a tether)
/* jshint node: true */

(function () {
    "use strict";

    var runtime = require("datawire-quark-node");
    var ipify = require("./ipify");

    function Consumer() {}
    Consumer.prototype.consume = function (ip) {
        console.log("IP is " + ip);
        process.exit();
    };

    var m = new ipify.MyExternalIP(runtime, new Consumer());
})();
